src/Compilers/Server/VBCSCompilerTests/CompilerServerApiTest.cs

```diff
@@ -542,5 +542,24 @@ public async Task IncorrectServerHashReturnsIncorrectHashResponse()
                 Assert.Equal(BuildResponse.ResponseType.IncorrectHash, buildResponse.Type);
             }
         }
+
+        [ConditionalFact(typeof(WindowsDesktopOnly))]
+        [WorkItem(33452, "https://github.com/dotnet/roslyn/issues/33452")]
+        public void QuotePipeName_Desktop()
+        {
+            var serverInfo = BuildServerConnection.GetServerProcessInfo(@"q:\tools", "name with space");
+            Assert.Equal(@"q:\tools\VBCSCompiler.exe", serverInfo.processFilePath);
+            Assert.Equal(@"q:\tools\VBCSCompiler.exe", serverInfo.toolFilePath);
+            Assert.Equal(@"""-pipename:name with space""", serverInfo.commandLineArguments);
+        }
+
+        [ConditionalFact(typeof(CoreClrOnly))]
+        [WorkItem(33452, "https://github.com/dotnet/roslyn/issues/33452")]
+        public void QuotePipeName_CoreClr()
+        {
+            var serverInfo = BuildServerConnection.GetServerProcessInfo(@"q:\tools", "name with space");
+            Assert.Equal(@"q:\tools\VBCSCompiler.dll", serverInfo.toolFilePath);
+            Assert.Equal(@"exec ""q:\tools\VBCSCompiler.dll"" ""-pipename:name with space""", serverInfo.commandLineArguments);
+        }
```

Comment:
Can the pipe name contain quotes? If so, consider adding a test for that case.

Reply:
No. The pipename is a combination of `%USERNAME%`, a boolean and a base64 encoded string. None of those can contain a quote. There are a few interesting symbols that can come out of the base64 string but they are already substituted out.